pkg/service/backup/purger.go

```diff
@@ -379,44 +379,33 @@ func (p purger) deleteFiles(ctx context.Context, location Location, files fileSe
 		missing atomic.Int64
 	)
 
-	defer func() {
-		s, m := int(success.Load()), int(missing.Load())
-		p.onDelete(ctx, total, s, m)
-	}()
-
 	// We cap the nr. of dirs purged in parallel to avoid opening expressive TCP connections in case there is no idle connection to host.
 	// maxParallelDirs is aligned with MaxIdleConnsPerHost in scyllaclient.DefaultTransport.
 	const maxParallelDirs = 100
 
 	f := func(i int) error {
-		var (
-			dir  = dirs[i]
-			rerr error
-		)
+		dir := dirs[i]
+		const maxBatchSize = 1000
```

Comment:
How did you choose this value of 1000 ?

Reply:
I wanted to limit batch size, so that we don't send request with millions of file name, as it could create problems with transport and timeouts. 1000 seems like it's small enough not to negatively impact the transport, and deleting 1000 files shouldn't take too long as well. What do you think about this number? 

On the other hand, I could make this API call asynchronous and use long polling for checking it's progress.
This would resolve the problem with timeout, but it would still be better not to delete all the files in a single batch anyway.